kroxylicious-api/src/test/java/io/kroxylicious/proxy/filter/SafeInvokerTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.kafka.common.message.ApiVersionsRequestData;
+import org.apache.kafka.common.message.ApiVersionsResponseData;
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.message.ResponseHeaderData;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import static org.apache.kafka.common.protocol.ApiKeys.API_VERSIONS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyShort;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class SafeInvokerTest {
```

Comment:
kudos. Really easy to read what the tests are about with the given when then notation.